omnigibson/object_states/toggle.py

```diff
@@ -51,16 +52,21 @@ def _initialize(self):
         pre_existing_mesh = get_prim_at_path(mesh_prim_path)
         # Create a primitive mesh if it doesn't already exist
         if not pre_existing_mesh:
-            button = UsdGeom.Sphere.Define(og.sim.stage, Sdf.Path(mesh_prim_path))
             self.radius = m.DEFAULT_RADIUS if self.radius is None else self.radius
+            mesh = create_primitive_mesh(
+                prim_path=mesh_prim_path,
+                primitive_type="Sphere",
+                extents=1.0,
+            )
         else:
             # Infer radius from mesh if not specified as an input
-            self.radius = pre_existing_mesh.GetAttribute("radius").Get() if self.radius is None else self.radius
+            recompute_extents(prim=pre_existing_mesh)
+            scale = np.array(pre_existing_mesh.GetAttribute("xformOp:scale").Get())
+            self.radius = np.cbrt(np.product(scale)) / 2.0 if self.radius is None else self.radius
 
         # Create the visual geom instance referencing the generated mesh prim
         self.visual_marker = VisualGeomPrim(prim_path=mesh_prim_path, name=f"{self.obj.name}_visual_marker")
-        self.visual_marker.set_attribute("radius", self.radius)
-        recompute_extents(prim=self.visual_marker.prim)
+        self.visual_marker.scale = self.radius * 2.0
```

Comment:
Oh I kind of see what you are trying to do here. So if the pre-existing mesh is non-uniformly scaled, we do the `np.cbrt(np.product(scale))` trick and force it to be uniformly scaled? This is also to prevent the physx warning, right? Maybe add a comment here to elaborate? I think it's a bit opaque currently.

Reply:
well now that we're using a mesh, we can scale stuff arbitrarily.

It's mainly to unify with the input kwarg `radius`, which is expected to be a scalar, not a 3-array. I suppose the reverse could also be done though, where `radius` can be a 3-array or a single number and we'd just apply it uniformly to the scale in the latter case.

But in general, I think can be a bit hard to infer scaling in xyz since it will be in the mesh's local frame -- i.e.: it might be difficult to figure out how the generated mesh will actually scale if you specify 3 different values for x,y,z. A single value for the radius is a lot more interpretable

Reply:
hmmm I am not sure we should do this. 

Here is my suggestion:
```
        if not pre_existing_mesh:
            mesh = create_primitive_mesh(
                prim_path=mesh_prim_path,
                primitive_type="Sphere",
                extents=1.0,
            )
        else:
            # CAN YOU ADD A COMMENT HERE EXPLAINING WHY THIS IS NEEDED?
            recompute_extents(prim=pre_existing_mesh)

        # Create the visual geom instance referencing the generated mesh prim
        self.visual_marker = VisualGeomPrim(prim_path=mesh_prim_path, name=f"{self.obj.name}_visual_marker")
        if not pre_existing_mesh:
            self.radius = m.DEFAULT_RADIUS if self.radius is None else self.radius
            self.visual_marker.scale = self.radius * 2.0
        self.visual_marker.initialize()
```

Reply:
agreed -- but i thought for now we would always have `pre_existing_mesh` but a sphere?

Reply:
alternatively, we can have the input be `scale` so it's shape-agnostic

Reply:
Right, I think I am just not a big fan of a mixture of things. If we have `self.radius`, let's make sure the mesh is actually a sphere (is there way to assert that?). Otherwise, we shouldn't have `self.radius` at all. Should be `self.scale` instead or `self.extent` or something like that.

Reply:
yeah, we can't assert a mesh is a sphere. we should just have self.extent / scale. I will make this update

Reply:
done